app/src/sharedTest/java/org/oppia/android/app/profile/ProfileChooserFragmentTest.kt

```diff
@@ -712,6 +717,191 @@ class ProfileChooserFragmentTest {
+
+      // Click twice to scroll to the end of the list.
+      onView(withId(R.id.profile_list_scroll_right)).perform(click())
+      testCoroutineDispatchers.runCurrent()
+
+      onView(withId(R.id.profile_list_scroll_right)).perform(click())
+      testCoroutineDispatchers.runCurrent()
+
+      // Click twice to scroll to the beginning of the list.
+      onView(withId(R.id.profile_list_scroll_left)).perform(click())
+      testCoroutineDispatchers.runCurrent()
+
+      onView(withId(R.id.profile_list_scroll_left)).perform(click())
+      testCoroutineDispatchers.runCurrent()
+
+      it.onActivity {
+        onView(
+          withViewAtPosition(
+            recyclerViewId = R.id.profiles_list_landscape,
+            position = 0,
+            targetViewId = R.id.profile_name_text,
+            itemMatcher = allOf(withText("Admin"), isDisplayed())
+          )
+        )
+        testCoroutineDispatchers.runCurrent()
+      }
+    }
+  }
+
+  @Test
+  @Config(qualifiers = "land")
+  fun testFragment_enableOnboardingV2_rtl_checkListIsAlphabetical() {
+    TestPlatformParameterModule.forceEnableOnboardingFlowV2(true)
+    profileTestHelper.addOnlyAdminProfile()
+    profileTestHelper.addMoreProfiles(9)
+
+    launch(ProfileChooserActivity::class.java).use {
+      it.onActivity { activity ->
+        activity.window.decorView.layoutDirection = ViewCompat.LAYOUT_DIRECTION_RTL
+        testCoroutineDispatchers.runCurrent()
+
+        onView(
+          withViewAtPosition(
+            recyclerViewId = R.id.profiles_list_landscape,
+            position = 0,
+            targetViewId = R.id.profile_name_text,
+            itemMatcher = allOf(withText("Admin"), isDisplayed())
+          )
+        )
+        testCoroutineDispatchers.runCurrent()
+      }
+    }
+  }
+
+  @Test
+  @Config(qualifiers = "land")
+  fun testFragment_enableOnboardingV2_rtl_checkLeftArrowScrollBehaviourIscorrect() {
+    TestPlatformParameterModule.forceEnableOnboardingFlowV2(true)
+    profileTestHelper.addOnlyAdminProfile()
+    profileTestHelper.addMoreProfiles(9)
+    launch(ProfileChooserActivity::class.java).use {
+      testCoroutineDispatchers.runCurrent()
+
+      // Click twice to scroll to the end of the list.
+      onView(withId(R.id.profile_list_scroll_left)).perform(click())
+      testCoroutineDispatchers.runCurrent()
+
+      onView(withId(R.id.profile_list_scroll_left)).perform(click())
+      testCoroutineDispatchers.runCurrent()
+
+      it.onActivity {
+        onView(
+          withViewAtPosition(
+            recyclerViewId = R.id.profiles_list_landscape,
+            position = 0,
+            targetViewId = R.id.profile_name_text,
+            itemMatcher = allOf(withText("I"), isDisplayed())
+          )
+        )
+
+        testCoroutineDispatchers.runCurrent()
+      }
+    }
+  }
+
+  @Test
+  @Config(qualifiers = "land")
+  fun testFragment_enableOnboardingV2_rtl_checkRightArrowScrollBehaviourIscorrect() {
```

Comment:
Are this & the one above actually verifying RTL? Only testFragment_enableOnboardingV2_rtl_checkListIsAlphabetical seems to be enabling RTL explicity.

Reply:
I refactored this test to use the Arabic locale.